src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -1087,7 +1087,7 @@ private ChainedHeader CreateAndValidateNewChainedHeader(BlockHeader currentBlock
             if (this.invalidHashesStore.IsInvalid(newHeaderHash))
             {
                 this.logger.LogTrace("(-)[HEADER_HASH_MARKED_INVALID]");
-                ConsensusErrors.BannedHash.Throw();
+                throw new HeaderInvalidException();
```

Comment:
Perhaps pass the "HEADER_HASH_MARKED_INVALID" message to the exception?

Reply:
No this is a specific exception.